src/main/java/lt/vu/trip/service/trip/TripServiceImpl.java

```diff
@@ -1,24 +1,64 @@
 package lt.vu.trip.service.trip;
 
 import lt.vu.trip.entity.Trip;
+import lt.vu.trip.entity.TripParticipationStatus;
+import lt.vu.trip.entity.User;
 import lt.vu.trip.repository.TripRepository;
+import lt.vu.trip.service.user.UserService;
 import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.PageRequest;
+import org.springframework.data.domain.Pageable;
+import org.springframework.data.domain.Sort;
 import org.springframework.stereotype.Component;
 
-import java.util.List;
-
 @Component
 public class TripServiceImpl implements TripService {
 
 	@Autowired
 	private TripRepository repo;
 
-	public List<Trip> getAll() {
-		return repo.findAll();
+	@Autowired
+	private UserService userService;
+
+	public Page<Trip> getAll(int page, int resultsPerPage, TripSearchCriteria criteria) {
+		return this.getList(page, resultsPerPage, criteria);
+	}
+
+	public Page<Trip> getOrganizedByCurrentUser(int page, int resultsPerPage, TripSearchCriteria criteria) {
+		User currentUser = userService.getCurrentUser();
+		criteria.setOrganizerId(currentUser.getId());
+		return this.getList(page, resultsPerPage, criteria);
+	}
+
+	public Page<Trip> getCurrentUserParticipatingIn(int page, int resultsPerPage, TripSearchCriteria criteria) {
```

Comment:
getCurrentUserParticipatingIn and getCurrentUserInvitedIn could be merged to a single method which would have an additional parameter for TripParticipationStatus. Currently it looks like almost duplicated code.